pandas/core/dtypes/dtypes.py

```diff
@@ -586,6 +602,10 @@ def __eq__(self, other):
                 self.unit == other.unit and
                 str(self.tz) == str(other.tz))
 
+    def __getstate__(self):
```

Comment:
this is  necessary?

Reply:
Yeah. For pickle compatibility. Else we try to pass `units=None, tz=None` to the constructor. We actually used to allow that, but it would let you create invalid Dtypes.

```python
In [3]: tz = pd.core.dtypes.dtypes.DatetimeTZDtype(tz='utc')

In [4]: tz.unit
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-a4f5c8783719> in <module>()
----> 1 tz.unit

AttributeError: 'DatetimeTZDtype' object has no attribute 'unit'

```

Reply:
maybe add this to the base class?

Reply:
I think I've fixed it properly now.

PandasExtensionType defines `__getstate__`, so we should define `__setstate__` to handle unpicking (if we aren't using `__new__`).

Reply:
makes sense